modules/checkbox/react/lib/Checkbox.tsx

```diff
@@ -251,7 +252,7 @@ export default class Checkbox extends React.Component<CheckboxProps> {
           <CheckboxInput
             checked={checked}
             disabled={disabled}
-            id={id}
+            id={id || this.id}
```

Comment:
Nit: when we destructure `id` out of `props` (L240) we should just assign `this.id` as a default value. This way if this logic ever needs to change you're not changing it in two places (here and L270).